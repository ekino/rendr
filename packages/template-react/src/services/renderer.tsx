import React from "react";
import { BlockRegistry, BlockRenderer, ContainerRenderer } from "../types";

export function createBlockRenderer(
  blockRegistry: BlockRegistry
): BlockRenderer {
  return function blockRenderer(block, key) {
    const { component: Component, settings: props } = blockRegistry(
      block.type,
      block.settings
    );

    return <Component key={key} {...props} blockRenderer={blockRenderer} />;
  };
}

export function createContainerRenderer(
  renderBlock: BlockRenderer
): ContainerRenderer {
  return (name, blocks) => {
    return (
      <>
        {blocks
          .filter(block => name === block.container)
          .map((block, i) => {
            return renderBlock(block, `k_${i}`);
          })}
      </>
    );
  };
}
