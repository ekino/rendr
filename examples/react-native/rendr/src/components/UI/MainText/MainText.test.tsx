import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

// Components.
import MainText from './MainText';

describe('<MainText>', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const tree = renderer.create(
        <MainText style={{color: 'black'}}>Testing</MainText>,
      );
      expect(tree).toMatchSnapshot();
    });
  });
});