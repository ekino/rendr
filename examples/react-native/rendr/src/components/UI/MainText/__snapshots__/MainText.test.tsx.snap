// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainText> Rendering should match to snapshot 1`] = `
<Text
  allowFontScaling={false}
  style={
    Object {
      "color": "black",
    }
  }
>
  Testing
</Text>
`;
